BitList.cs

```diff
@@ -0,0 +1,1496 @@
+        /// <param name="index">Start index.</param>
+        /// <param name="count">Count of bits.</param>
+        /// <exception cref="ArgumentOutOfRangeException" />
+        public void RemoveRange(int index, int count)
+        {
+            if (!IsEmpty && index < bits.Length && index >= 0 && index + count <= bits.Length - index && count > 0)
+            {
+                var bools = ToBooleanArray().ToList();
+                bools.RemoveRange(index, count);
+                bits = new BitArray(bools.ToArray());
+            }
+            else
+                throw new ArgumentOutOfRangeException();
+        }
+
+        /// <summary>
+        /// Remove the specified bit from this instance.
+        /// </summary>
+        public bool Remove(bool item)
+        {
+            try
+            {
+                RemoveAt(IndexOf(item));
+                return true;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Remove the specified list of bit from this instance.
+        /// </summary>
+        /// <exception cref="EmptyInstanceException" />
+        public bool Remove(BitList items)
+        {
+            if (IsEmpty)
+                throw new EmptyInstanceException();
+            else
+            {
+                var thisBitList = ToText();
+                var thoseBitList = items.ToText();
+                if (!thisBitList.Contains(thoseBitList))
+                    return false;
+                var newBitList = thisBitList.Replace(thoseBitList, string.Empty);
+                bits = Parse(newBitList).bits;
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Get the index of a bit in this instance.
+        /// </summary>
+        /// <param name="bit">A bit.</param>
+        /// <returns>The <paramref name="item"/> index.</returns>
+        public int IndexOf(bool item)
+        {
+            for (int i = 0; i < Count; i++)
+            {
+                if (item == this[i])
+                    return i;
+            }
+            return -1;
+        }
+
+        /// <summary>
+        /// Get the index of a list of bit in this instance.
+        /// </summary>
+        /// <param name="items">A list of bit.</param>
+        public int IndexOf(BitList items)
+        {
+            for (int i = 0; i < Count; i++)
+            {
+                if (items == this[i, items.Count])
+                    return i;
+            }
+            return -1;
+        }
+
+        /// <summary>
+        /// Shift bits to left.
+        /// </summary>
+        public void LeftShift(int count)
+        {
+            if (count != 0)
+            {
+                for (int i = 1; i == Math.Abs(count); i++)
+                {
+                    if (count > 0)
+                        Add(false);
+                    else
+                    {
+                        try
```

Comment:
Codacy found an issue: [Refactor this code to not nest more than 3 control flow statements.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)